<TITLE>fastacmd README</TITLE> 
<BODY bgcolor="FFFFFF" link="0000FF" vlink="ff0000" text="000000" >  
<h1 align = center>FASTACMD HELP MANUAL</h1>
<HR>


<h3>Table of Contents</h3>
<PRE> 
    <a href = #21>Introduction</a>
 
    <a href = #25>Command line options</a>
 
    <a href = #70>Usage</a>
 
 </PRE>
 
 
<h3><a name = 21></a>Introduction</h3>

<PRE> 
fastacmd retrives FASTA formatted sequences from a blast database, as long as
it it was successfully formatted using the '-o' option.  
</PRE> 

<h3><a name = 25></a>Command line options</h3>
<PRE>
The fastacmd options are:
 
fastacmd 2.2.10   arguments:
 
  -d  Database [String]  Optional
    default = nr
  -p  Type of file
         G - guess mode (look for protein, then nucleotide)
         T - protein   
         F - nucleotide [String]  Optional
    default = G
  -s  Search string: GIs, accessions and loci may be used delimited
      by comma. [String]  Optional
  -i  Input file wilth GIs/accessions/loci for batch
      retrieval [String]  Optional
  -a  Retrieve duplicate accessions [T/F]  Optional
    default = F
  -l  Line length for sequence [Integer]  Optional
    default = 80
  -t  Definition line should contain target gi only [T/F]  Optional
    default = F
  -o  Output file [File Out]  Optional
    default = stdout
  -c  Use Ctrl-A's as non-redundant defline separator [T/F]  Optional
    default = F
  -D  Dump the entire database as (default is not to dump anything):
      1 FASTA
      2 Gi list
 [Integer]  Optional
    default = 0
  -L  Range of sequence to extract (Format: start,stop)
      0 in 'start' refers to the beginning of the sequence
      0 in 'stop' refers to the end of the sequence [String]  Optional
    default = 0,0
  -S  Strand on subsequence (nucleotide only): 1 is top, 2 is bottom [Integer]
    default = 1
  -T  Print taxonomic information for requested sequence(s) [T/F]
    default = F
  -I  Print database information only (overrides all other options) [T/F]
    default = F
  -P  Retrieve sequences with this PIG [Integer]  Optional
</PRE>
 
<h3><a name = 70></a>Usage</h3>
<PRE>
 
1.) Retrieving a sequence by identifier:
 
fastacmd -d nt -s 555
&gt;gi|555|emb|X65215.1|BTMISATN B.taurus microsatellite DNA (624bp)
ACCTCCACTAGCTTTGTTTGTAGTGATGCTCTGTAGCACCACTGGGAAGCCCTTTAATGAATGTGCCTTTCCGCAAATCA
CACACACACAAATACACTTATAGAAACAAGGTGATTTTCTTGAAATAATAAAACAAAATTTGGAAGAAGATTTTTACTGT
CTTAGGAAAAGTAAGGCATTGGAAGGTGGCTAGGTATGACATATGAAGTTGCATTTTAAAACTGGAATTGGACAACTGAT
ATTCAGTGATATTTATGCTACTACCTTCTAGAATCGAGAGCATGCACCCCACTCTGTACTCTTGCCTGGAGAATCCATGA
TGAGAGCCTGGTAGGCTGCAGTCCATGGGGTCACACAGAGTCGGACATGACTGAGCGACTTCACTTTCACTTTTCAATTT
CATGCATTGGAGCCGGAAATGGCAACCCACTCCAGTGTTCTTGCCTGGAGAATCCCAGGGATGGGGAAGCCTGGTGGGCT
GCTGTCTATGGGGTCGCAGAGAGTCAGACACGACTGAAGTGACTTAGCAGCAACCTTCTGGAATAAACGCCTCAGGCTTT
AAACTCTGGCTTGACCATTCACTAGCCATGGGATCCACTAGAGTCGACCTGCAGGCATGCAAGC
 
If the identifier is not a gi or an accession, you must pass the entire seqid
string to fastacmd. That is, if your sequence is
 
&gt;gnl|mydb|myid my sequence description
ACGT...
 
, you must search for it with
 
fastacmd -d mydb -s 'gnl|mydb|myid'
 
2.) Obtaining a FASTA file from a blast database:
 
fastacmd -D 1 -d nt -o nt.fsa
[output removed for brevity]
 
3.) Retrieving only part of a sequence:
 
fastacmd -d nt -s 555 -L0,32
gi|555:1-32 B.taurus microsatellite DNA (624bp)
ACCTCCACTAGCTTTGTTTGTAGTGATGCTCT
 
Return values
-------------
The following exit values are returned:
 
     0     Completed successfully
 
     1     An error occurred
 
     2     Blast database was not found
 
     3     Failed search (accession, gi, taxonomy info)
 
     4     No taxonomy database was found
 
</PRE> 

